IBF-Typhoon-model/lib/pipeline.py

```diff
@@ -0,0 +1,314 @@
+#!/bin/sh
+
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Oct 31 16:01:00 2020
+
+@author: ATeklesadik
+"""
+import sys
+import os
+from datetime import datetime, timedelta
+from sys import platform
+import subprocess
+import logging
+import traceback
+
+import pandas as pd
+from pybufrkit.decoder import Decoder
+import numpy as np
+from geopandas.tools import sjoin
+import geopandas as gpd
+import click
+
+# Set up logger
+logging.root.handlers = []
+logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.DEBUG, filename='ex.log')
+# set up logging to console
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+# set a format which is simpler for console use
+formatter = logging.Formatter('%(asctime)s : %(levelname)s : %(message)s')
+console.setFormatter(formatter)
+logging.getLogger("").addHandler(console)
+
+decoder = Decoder()
+#path='C:/Users/ATeklesadik/OneDrive - Rode Kruis/Documents/Typhoon-Impact-based-forecasting-model/IBF-Typhoon-model/'
+#path='/home/fbf/'
+path = './'
+#%%
+sys.path.insert(0, path+'lib')
+os.chdir(path)
+from settings import fTP_LOGIN, fTP_PASSWORD, uCL_USERNAME, uCL_PASSWORD
+from climada.hazard import Centroids, TropCyclone,TCTracks
+from climada.hazard.tc_tracks import estimate_roci,estimate_rmw
+from climada.hazard.tc_tracks_forecast import TCForecast
+from utility_fun import track_data_clean,Check_for_active_typhoon,Sendemail,ucl_data
+from utility_fun import Rainfall_data
+
+
+@click.command()
+@click.option('--path', default='./', help='main directory')
+@click.option('--remote_dir_', default='20210421120000', help='remote directory for ECMWF forecast data') 
+@click.option('--active_typhoon', default='SURIGAE',help='name for active typhoon')
+
+def main(path,remote_dir_,active_typhoon):
+    print('---------------------AUTOMATION SCRIPT STARTED---------------------------------')
+    print(str(datetime.now()))
+    #%% check for active typhoons
+    print('---------------------check for active typhoons---------------------------------')
+    print(str(datetime.now()))
+    Activetyphoon=Check_for_active_typhoon.check_active_typhoon()
+    if Activetyphoon==[]:
+      remote_dir=remote_dir_#'20210421120000' #for downloading test data otherwise set it to None
+      Activetyphoon=[active_typhoon]  #name of typhoon for test
```

Comment:
It looks like the `active_typhoon` and `remote_dir` command line argument options would only be used in case there is no active typhoon. Is this what you intended?